torch/_inductor/compile_fx.py

```diff
@@ -563,6 +563,7 @@ def compile_fx_inner(
         # the counter here because we may dropped into compile_fx directly
         # from lazy backwards compilation.
         stack.enter_context(_WaitCounter("pytorch.wait_counter.dynamo_compile").guard())
+        stack.enter_context(torch._dynamo.callback_handler.install_callbacks())
```

Comment:
I mentioned in chat, we should only do this if is_backward is true, or else we will do this twice for the forward pass. This might be okay but it's better to be safe than sorry

Reply:
I created a new PR with a different approach to duplicate callbacks prevention and unit tests: https://github.com/pytorch/pytorch/pull/145732. 

Closing this one.